engine/overworld/map_setup.asm

```diff
@@ -69,12 +69,173 @@ SuspendMapAnims:
 	ldh [hMapAnims], a
 	ret
 
+LoadMapObjects_Connection:
+	call ReassociateMapObjectsOrDelete
 LoadMapObjects:
 	ld a, MAPCALLBACK_OBJECTS
 	call RunMapCallback
 	call LoadObjectMasks
 	farjp InitializeVisibleSprites
 
+ReassociateMapObjectsOrDelete:
+; Reassociates the objects if their coordinates match the relative coordinates
+; of an object in the new map relative to the player, or it deletes the object.
+	ld a, 1
+	ldh [hObjectStructIndexBuffer], a
+	ld bc, wObjectStructs + OBJECT_LENGTH
+	ld e, NUM_OBJECT_STRUCTS - 1
+.loop_objects
+	push de
+	ld hl, OBJECT_LAST_MAP_X
+	add hl, bc
+	ld a, [hli]
+	ld e, [hl]
+	ld d, a
+	call .CheckForMatchingMapObject
+	push af
+	call nc, .DeleteObjectStruct
+	pop af
+	jr nc, .continue
```

Comment:
How about this...
```suggestion
	jr nc, .DeleteObjectStruct
```

Reply:
We want to continue executing the loop.. not jump to delete and return out of the entire routine.